SOSS/dms/soss_utils.py

```diff
@@ -0,0 +1,164 @@
+    if shift >= 0:
+        result[shift:] = a[:-shift]
+    else:
+        result[:shift] = a[-shift:]
+
+    return result
+
+
+def robust_polyfit(x, y, order, maxiter=5, nstd=3.):
+    """Perform a robust polynomial fit.
+
+    :param x: x data to fit.
+    :param y: y data to fit.
+    :param order: polynomial order to use.
+    :param maxiter: number of iterations for rejecting outliers.
+    :param nstd: number of standard deviations to use when rejecting outliers.
+
+    :type x: array[float]
+    :type y: array[float]
+    :type order: int
+    :type maxiter: int
+    :type nstd: float
+
+    :returns: param - best-fit polynomial parameters.
+    :rtype: array[float]
+    """
+
+    mask = np.ones_like(x, dtype='bool')
+    for niter in range(maxiter):
+
+        # Fit the data and evaluate the best-fit model.
+        param = np.polyfit(x[mask], y[mask], order)
+        yfit = np.polyval(param, x)
+
+        # Compute residuals and mask ouliers.
+        res = y - yfit
+        stddev = np.std(res)
+        mask = np.abs(res) <= nstd*stddev
+
+    return param
+
+
+def get_image_dim(image, header=None, verbose=False):
+    """Determine the properties of the image array.
+
+    :param image: A 2D image of the detector.
+    :param header: The header from one of the SOSS reference files.
+    :param verbose: If set True some diagnostic plots will be made.
+
+    :type image: array[float]
+    :type header: astropy.io.fits.Header
+    :type verbose: bool
+
+    :returns:
+    dimx, dimy
+        The dimensions of the stack array.
+    xos, yos
+        The oversampling factors of the stack array.
+    xnative, ynative
+        The dimensions of the stack image, in native pixels.
+    padding
+        Amount of padding around the image, in native pixels.
+    refpix_mask
+        Boolean array indicating which pixels are lightsensitive (True) and which are reference pixels (False).
+
+    :rtype: Tuple(int, int, int, int, int, int, int, array[bool])
+    """
+
+    # Dimensions of the subarray.
+    dimy, dimx = np.shape(image)
+
+    # If no header was passed we have to check all possible sizes.
+    if header is None:
+
+        # Initialize padding to zero in this case because it is not a reference file.
+        padding = 0
+
+        # Assume the stack is a valid SOSS subarray.
+        # FULL: 2048x2048 or 2040x2040 (working pixels) or multiple if oversampled.
+        # SUBSTRIP96: 2048x96 or 2040x96 (working pixels) or multiple if oversampled.
+        # SUBSTRIP256: 2048x256 or 2040x252 (working pixels) or multiple if oversampled.
+
+        # Check if the size of the x-axis is valid.
+        if (dimx % 2048) == 0:
+            xnative = 2048
+            xos = int(dimx // 2048)
+
+        elif (dimx % 2040) == 0:
+            xnative = 2040
+            xos = int(dimx // 2040)
+
+        else:
+            raise ValueError('Stack X dimension has unrecognized size of {:}. Accepts 2048, 2040 or multiple of.'.format(dimx))
```

Comment:
long line can easily split this up 

```python
errormsg = 'My error message'
raise ValueError(errormsg)
```

Reply:
It's still too long for the Stack Y one.

Reply:
There's always a way

```python
errormsg =  ('Stack X dimension has unrecognized size of {:}.'
             'Accepts 2048, 2040 or multiple of.')
raise ValueError(errormsg.format(dimx))
```

Reply:
Changed.